dubbo-config/dubbo-config-api/src/main/java/com/alibaba/dubbo/config/ServiceConfig.java

```diff
@@ -491,7 +491,7 @@ private void doExportUrlsFor1Protocol(ProtocolConfig protocolConfig, List<URL> r
                 }
                 if (registryURLs != null && !registryURLs.isEmpty()) {
                     for (URL registryURL : registryURLs) {
-                        url = url.addParameterIfAbsent("dynamic", registryURL.getParameter("dynamic"));
+                        url = url.addParameterIfAbsent(Constants.DYNAMIC_KEY, registryURL.getParameter("dynamic"));

```

Comment:
Why is the string in `registryURL.getParameter("dynamic")` not getting replaced with `Constants.DYNAMIC_KEY`?

Reply:
I think key should be managed by Constants, Instead of writing directly to the code. Like this ,Post-manitenance comfortably.

Reply:
Yes, I agree with you. My question is, why it is not some sth. like this:

```java
url = url.addParameterIfAbsent(Constants.DYNAMIC_KEY, registryURL.getParameter(Constants.DYNAMIC_KEY));
```

Reply:
emmm... you are right, I will update it.